x/pool-incentives/keeper/distr_test.go

```diff
@@ -0,0 +1,291 @@
+
+	pot2, err = suite.app.IncentivesKeeper.GetPotByID(suite.ctx, pot2Id)
+	suite.NoError(err)
+	suite.Equal("49999stake", pot2.Coins.String())
+
+	pot3, err = suite.app.IncentivesKeeper.GetPotByID(suite.ctx, pot3Id)
+	suite.NoError(err)
+	suite.Equal("75000stake", pot3.Coins.String())
+
+	// ------------ test community pool distribution when potId is zero ------------ //
+	// Mint more stake coin to the fee collector.
+	err = suite.app.BankKeeper.AddCoins(
+		suite.ctx,
+		suite.app.AccountKeeper.GetModuleAddress(authtypes.FeeCollectorName),
+		sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(100000))),
+	)
+	suite.NoError(err)
+
+	// record original community pool balance
+	feePoolOrigin := suite.app.DistrKeeper.GetFeePool(suite.ctx)
+
+	// Create 3 records including community pool
+	err = keeper.UpdateDistrRecords(suite.ctx, types.DistrRecord{
+		PotId:  pot1Id,
+		Weight: sdk.NewInt(100),
+	}, types.DistrRecord{
+		PotId:  pot2Id,
+		Weight: sdk.NewInt(200),
+	}, types.DistrRecord{
+		PotId:  0,
+		Weight: sdk.NewInt(700),
+	})
+	suite.NoError(err)
+
+	// In this time, there are 3 records, so the assets to be allocated to the pots proportionally.
+	err = keeper.AllocateAsset(suite.ctx, sdk.NewCoin("stake", sdk.NewInt(100000)))
+	suite.NoError(err)
+
+	// check community pool balance increase
+	feePoolNew := suite.app.DistrKeeper.GetFeePool(suite.ctx)
+	suite.Equal(feePoolOrigin.CommunityPool.Add(sdk.NewDecCoin("stake", sdk.NewInt(70000))), feePoolNew.CommunityPool)
+
+	// ------------ test community pool distribution when no distribution records are set ------------ //
+	// Mint more stake coin to the fee collector.
+	err = suite.app.BankKeeper.AddCoins(
+		suite.ctx,
+		suite.app.AccountKeeper.GetModuleAddress(authtypes.FeeCollectorName),
+		sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(100000))),
+	)
+	suite.NoError(err)
+
+	// record original community pool balance
+	feePoolOrigin = suite.app.DistrKeeper.GetFeePool(suite.ctx)
+
+	// set empty records set
+	err = keeper.UpdateDistrRecords(suite.ctx)
+	suite.NoError(err)
+
+	// In this time, all should be allocated to community pool
+	err = keeper.AllocateAsset(suite.ctx, sdk.NewCoin("stake", sdk.NewInt(100000)))
+	suite.NoError(err)
+
+	// check community pool balance increase
+	feePoolNew = suite.app.DistrKeeper.GetFeePool(suite.ctx)
+	suite.Equal(feePoolOrigin.CommunityPool.Add(sdk.NewDecCoin("stake", sdk.NewInt(100000))), feePoolNew.CommunityPool)
+}
+
+func (suite *KeeperTestSuite) TestUpdateDistrRecords() uint64 {
+	suite.SetupTest()
+
+	keeper := suite.app.PoolIncentivesKeeper
+
+	// Not existent pot.
+	err := keeper.UpdateDistrRecords(suite.ctx, types.DistrRecord{
+		PotId:  1,
+		Weight: sdk.NewInt(100),
+	})
+	suite.NoError(err)
+
+	poolId := suite.preparePool()
+
+	// LockableDurations should be 1, 3, 7 hours from the default genesis state.
+	lockableDurations := keeper.GetLockableDurations(suite.ctx)
+	suite.Equal(3, len(lockableDurations))
```

Comment:
Can we test exact equality with expected here? (By making a struct with whats expected, or from reading the expected from wherever its defined?)